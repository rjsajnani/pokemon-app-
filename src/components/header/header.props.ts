export type HeaderProps = {
  image: string;
  backgroundColor: string;
};
