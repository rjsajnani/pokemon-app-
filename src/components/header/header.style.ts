import { StyleSheet } from 'react-native';

export const HeaderStyle = StyleSheet.create({
  container: {
    overflow: 'hidden',
  },
  bgStyles: {
    width: '100%',
    height: 280,
    position: 'absolute',
    borderBottomEndRadius: 300,
    borderBottomLeftRadius: 300,
    transform: [{ scaleX: 2 }],
    backgroundColor: 'black',
  },
  name: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 27,
  },
  order: {
    color: '#fff',
    fontWeight: 'bold',
  },
  imageContainer: {
    paddingTop: 80,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  image: {
    width: 250,
    height: 250,
  },
});
