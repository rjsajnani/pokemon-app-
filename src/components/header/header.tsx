import { Image, View } from 'react-native';
import { HeaderStyle as styles } from './header.style';
import { HeaderProps as Props } from './header.props';

export const Header: React.FC<Props> = (props) => {
  const { image, backgroundColor } = props;

  return (
    <View style={styles.container}>
      <View style={{ ...styles.bgStyles, backgroundColor: backgroundColor }} />
      <View style={styles.imageContainer}>
        <Image
          source={{
            uri: image,
          }}
          style={styles.image}
        />
      </View>
    </View>
  );
};
