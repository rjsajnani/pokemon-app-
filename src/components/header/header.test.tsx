import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';

import { Header } from './header';

describe('<Header />', () => {
  let instance: RenderAPI;
  beforeEach(function () {
    instance = render(
      <Header
        image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png"
        backgroundColor="#A8A77A"
      />,
    );
  });

  it('renders correctly', () => {
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
