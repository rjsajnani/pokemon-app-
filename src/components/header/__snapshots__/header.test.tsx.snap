// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#A8A77A",
        "borderBottomEndRadius": 300,
        "borderBottomLeftRadius": 300,
        "height": 280,
        "position": "absolute",
        "transform": [
          {
            "scaleX": 2,
          },
        ],
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "flex-end",
        "paddingTop": 80,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png",
        }
      }
      style={
        {
          "height": 250,
          "width": 250,
        }
      }
    />
  </View>
</View>
`;
